src/Components/WebAssembly/Server/src/TargetPickerUi.cs

```diff
@@ -366,9 +366,34 @@ private string GetDevToolsUrlWithProxy(BrowserTab tabToDebug)
     {
         var underlyingV8Endpoint = new Uri(tabToDebug.WebSocketDebuggerUrl);
         var proxyEndpoint = new Uri(_debugProxyUrl);
-        var devToolsUrlAbsolute = new Uri(_browserHost + tabToDebug.DevtoolsFrontendUrl);
+        var devToolsUrlAbsolute = new Uri(new Uri(_browserHost), relativeUri: NormalizeDevtoolsFrontendUrl(tabToDebug.DevtoolsFrontendUrl));
```

Comment:
Is the scheme and host part of the `DevToolsFrontendUrl` different from those of `_browserHost`?

Could the logic here be something like `Uri.TryCreate(tabToDebug.DevtoolsFrontendUrl, UriKind.Absolute, out var devToolsAbsolute) ? devToolsAbsolute : new Uri(_browserHost + tabToDebug.DevtoolsFrontendUrl)` instead?

Reply:
Yeah, looking at where this is called [from](https://github.com/dotnet/aspnetcore/blob/main/src/Components/WebAssembly/Server/src/WebAssemblyNetDebugProxyAppBuilderExtensions.cs#L28) the host is either provided on the URL or hardcoded to different values. I believe the right thing to do is if we get here and we have an absolute URL, just use that, and if not fallback to the old logic.

Unless I'm missing something I don't understand why we would need to do more fiddling with the URL.

Reply:
I tried the absolute url returned but doesn't work.. Aren't local url (chrome point to `apphost.com` and edge point to `aka.ms` and even if the developer tools page opens in some way then goes in error due to websocket connection error